core/src/test/java/org/apache/iceberg/TestPartitioning.java

```diff
@@ -386,4 +386,26 @@ public void testGroupingKeyTypeWithIncompatibleSpecEvolution() {
         "Conflicting partition fields",
         () -> Partitioning.groupingKeyType(table.schema(), table.specs().values()));
   }
+
+  @Test
+  public void testDeletingPartitionField() {
+    TestTables.TestTable table =
+        TestTables.create(tableDir, "test", SCHEMA, BY_DATA_SPEC, V1_FORMAT_VERSION);
+
+    table.updateSpec().removeField("data").commit();
+
+    table.updateSchema().deleteColumn("data").commit();
+
+    // This would throw a NPE
```

Comment:
is this comment still apply?

Reply:
I agree that it is better to remove it, thanks!